src/app/AttributePathExpandIterator.cpp

```diff
@@ -127,11 +127,13 @@ void AttributePathExpandIterator::PrepareAttributeIndexRange(const AttributePath
             // and overflow to 0 for the max index) to us not going through
             // non-metadata global attributes for this attribute.
             mGlobalAttributeIndex = UINT8_MAX;
-            for (uint8_t idx = 0; idx < ArraySize(GlobalAttributesNotInMetadata); ++idx)
+            for (size_t idx = 0; idx < ArraySize(GlobalAttributesNotInMetadata); ++idx)
             {
                 if (GlobalAttributesNotInMetadata[idx] == aAttributePath.mAttributeId)
                 {
-                    mGlobalAttributeIndex = idx;
+                    // generally safe: we have a static cast check for GlobalAttributesNotInMetaData size
+                    VerifyOrDie(idx <= UINT8_MAX);
```

Comment:
I would prefer it if we cast the loop condition ArraySize bit to `uint8_t` and had a static assert there that ensures it's safe, and didn't touch the actual type of the index in this code.